poucave/app.py

```diff
@@ -303,18 +329,73 @@ async def _run_checks_parallel(checks, cache, tracker, force=False):
     return body
 
 
+def _send_sentry(event, payload):
+    """
+    Send a Sentry message when the check state changes.
+    """
+    check = payload["check"]
+    data = payload["result"]["data"]
+    success = payload["result"]["success"]
+
+    with configure_scope() as scope:
+        scope.set_extra("data", data)
+        scope.set_tag("source", "check")
+        # Group check failures (and not by message).
+        identifier = f"{check.project}/{check.name}"
+        scope.fingerprint = [identifier]
+    capture_message(
+        f"{identifier} " + ("recovered" if success else "is failing"),
+        level="info" if success else "error",
+    )
+
+
+def _log_result(event, payload):
+    """
+    Log check result data to stdout.
+
+    Our logging setup stores JSON MozLog output into BigQuery, and
+    this allows us to keep track of checks history.
+    """
+    check = payload["check"]
+    result = payload["result"]
+
+    infos = {
+        "time": utils.utcnow().isoformat(),
+        "project": check.project,
+        "check": check.name,
+        "tags": check.tags,
+        "success": result["success"],
+        # Convert result data to string (for type consistency).
+        "data": json.dumps(result["data"]),
+        # An optional scalar value (see below)
+        "plot": 0.0,
```

Comment:
Can we use `None` for no data here? It should be translated to `null` in JSON.

Reply:
Since BigQuery may create the column from the data of the first entry I thought that `null` could cause issues. Do we have more info about the code that takes the log entry and process it?

Reply:
This indeed sounds like a valid concern. I don't know what will happen. I think The JSON is simply ingested into StackDriver and Bigtable as it is, and any parsing happens on Google's side. A `null` value may very well result in a table with a string column, but maybe we can simply try it? I don't have a strong opinion either way.

Reply:
I agree that semantically this `0.0` here is bad.

I'm happy to try, and release a dot version if it fails. But you may have to get your hands dirty too (ie. drop the table) in case it does not get ingested :sweat:

Reply:
That's fine. Since a new table is created every day, I guess I can also fix it by just waiting for the next day. :)